x-pack/metricbeat/module/mssql/db/db.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package db
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/x-pack/metricbeat/module/mssql"
+)
+
+func init() {
+	mb.Registry.MustAddMetricSet("mssql", "db", New,
+		mb.DefaultMetricSet(),
+		mb.WithHostParser(mssql.HostParser))
+}
+
+// MetricSet type defines all fields of the MetricSet
+type MetricSet struct {
+	mb.BaseMetricSet
+	log     *logp.Logger
+	fetcher *mssql.Fetcher
+}
+
+// New create a new instance of the MetricSet
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The mssql db metricset is beta.")
+
+	logger := logp.NewLogger("mssql.db").With("host", base.HostData().SanitizedURI)
+
+	fetcher, err := mssql.NewFetcher(base.HostData().URI,
```

Comment:
Do you mean apart from this one? https://github.com/elastic/beats/pull/9414/files#diff-c4de2e39351be98f70b39ad4e9dfa233R58

Reply:
Is this being called at some moment? If you implement a `Close()` method in the metricset it will be called when the metricset is stopped, the close in the metricset could call `m.fetcher.Close()`.

Reply:
Now I understand. Somehow I was thinking that `Close` was being called only at the module level. Thanks for the clarification